src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -97,13 +97,12 @@ private void MakeCompilationOutputRunnableForCoreCLR()
         {   
             WriteDepsFileAndCopyProjectDependencies(_exporter);
 
-            var emitEntryPoint = _compilerOptions.EmitEntryPoint ?? false;
+            var isRunnable = _compilerOptions.EmitEntryPoint ?? _context.ProjectFile.OverrideIsRunnable;
 
-            if (emitEntryPoint && !_context.IsPortable)
+            if (isRunnable && !_context.IsPortable)
             {
-                // TODO: Pick a host based on the RID
                 VerifyCoreClrPresenceInPackageGraph();
-                CoreHost.CopyTo(_runtimeOutputPath, _compilerOptions.OutputName + Constants.ExeSuffix);
```

Comment:
I guess I don't see where the host is coming from in build. What's getting it from the dependencies?

---
In reply to: [71797819](https://github.com/dotnet/cli/pull/3911#discussion_r71797819) [](ancestors = 71797819,71797731,71797574)